src/docfx/lib/HtmlUtility.cs

```diff
@@ -215,13 +199,35 @@ private static string AddLocaleIfMissing(string href, string locale)
 
         private static int CountWordInText(string text)
         {
-            if (string.IsNullOrWhiteSpace(text))
+            var total = 0;
+            var word = false;
+
+            foreach (var ch in text)
             {
-                return 0;
+                if (ch == ' ' || ch == '\t' || ch == '\n')
+                {
+                    if (word)
+                    {
+                        word = false;
+                        total++;
+                    }
+                }
+                else if (
+                    ch != '.' && ch != '?' && ch != '!' &&
+                    ch != ';' && ch != ':' && ch != ',' &&
+                    ch != '(' && ch != ')' && ch != '[' &&
+                    ch != ']')
```

Comment:
can we put special chars into HashSet<char> hashset, then call set.Contains(ch) to get a better performance?

Reply:
it depends